### ISSUE TITLE
raise ValueError(data_type + ' for ' + key + ' is not a valid data type')

### ISSUE THREAD

**Issue by stoop4** [shicky/Tornado-Async-ODM#1]
This error happens all the time.

**Comment by shicky** [comment]
Data type verification is based on TYPE constant you define.
For example, let's say object is defined as following:
fields = {'_id': TYPE_EMAIL}
'_id' need to be in 'email format' or else this error would appear.

**Comment by stoop4** [comment]
Yes. But i think the type es TYPE_STR and the value is 'root'. And in log is written t for key is not a valid datatype. May i'm using it wrong.

**Comment by shicky** [comment]
This error indicates that the TYPE constant you are using cannot be found.
Perhapse you are using older version of the library.
Trying printing out TYPE_STR to see what value it has.
You need to import constant TYPE_STR from tornado_async_odm.
